src/domain/usecases/ExportUsersUseCase.ts

```diff
@@ -0,0 +1,154 @@
+import _ from "lodash";
+import moment from "moment";
+import Papa from "papaparse";
+import i18n from "../../locales";
+
+import { Future, FutureData } from "../entities/Future";
+import { User } from "../entities/User";
+import { ListOptions, UserRepository } from "../repositories/UserRepository";
+import { UseCase } from "../../CompositionRoot";
+import { OrgUnitKey } from "../entities/OrgUnit";
+
+const fieldSplitChar = "||";
+const defaultNameField = "name";
+const columnNameFromPropertyMapping = {
+    id: i18n.t("ID"),
+    username: i18n.t("Username"),
+    password: i18n.t("Password"),
+    name: i18n.t("Name"),
+    firstName: i18n.t("First name"),
+    surname: i18n.t("Surname"),
+    email: i18n.t("Email"),
+    phoneNumber: i18n.t("Phone number"),
+    lastUpdated: i18n.t("Updated"),
+    lastLogin: i18n.t("Last login"),
+    apiUrl: i18n.t("API URL"),
+    created: i18n.t("Created"),
+    userRoles: i18n.t("Roles"),
+    userGroups: i18n.t("Groups"),
+    organisationUnits: i18n.t("OUCapture"),
+    dataViewOrganisationUnits: i18n.t("OUOutput"),
+    searchOrganisationsUnits: i18n.t("OUSearch"),
+    disabled: i18n.t("Disabled"),
+    status: i18n.t("Status"),
+    openId: i18n.t("Open ID"),
+    createdBy: i18n.t("Created by"),
+    lastModifiedBy: i18n.t("Last modified by"),
+};
+
+export class ExportUsersUseCase implements UseCase {
```

Comment:
Let's remove the `implements UseCase` since most of the time this class is only going to have the execute method.